app/services/supplementary-billing/check-live-bill-run.service.js

```diff
@@ -0,0 +1,38 @@
+'use strict'
+
+/**
+ * Checks whether a "live" bill run exists for the specified region, scheme, type and financial year
+ * @module CheckLiveBillRunService
+ */
+
+const BillingBatchModel = require('../../models/water/billing-batch.model.js')
+
+const LIVE_STATUSES = ['processing', 'ready', 'review', 'queued']
+
+/**
+ * Check whether a "live" bill run exists for the specified region, scheme, type and financial year
+ *
+ * We define "live" as having the status `processing`, `ready`, `review` or `queued`
+ *
+ * @param {*} regionId The id of the region to be checked
+ * @param {*} financialYear The financial year to be checked
+ *
+ * @returns {Boolean} Whether a "live" bill run exists
+ */
+async function go (regionId, financialYear) {
+  const numberOfLiveBillRuns = await BillingBatchModel.query()
+    .where({
```

Comment:
It's a negligible tweak. But do you think there is value in adding a `select('billing_batch_id`)` to the query to limit the amount of data being returned?

We only care about the number of records returned, not the data in those records 🤷

Reply:
Yep that's fair enough, I somehow managed to miss the bit of the docs for `.resultSize()` that says it does actually run that query!